### PR TITLE
Exercise 3

### PR DESCRIPTION
**Pull request by slipstreamamx**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

**Comment by slipstreamamx** [comment]
I will complete the E2E testing during module 6 and 7.

File: tests/test_Integration.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+class StubResponse():
+    def __init__(self, fake_response_data):
+        self.fake_response_data = fake_response_data
+    
+    def json(self): 
+        return self.fake_response_data
+    
+def stub(url, params):
+    test_board_id = os.environ.get('TRELLO_BOARD_ID')
+    fake_response_data = None
```
Comment (BenRamchandani) [inline tests/test_Integration.py:27]: minor, but you don't need this variable

File: tests/test_Integration.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    }
+]
+
+sample_trello_card = {
+    "id": not_started_item_id,
+    "idShort": "2",
+    "name": "Task 2",
+    "desc": "Task 2 not started",
+    "due": "2022-05-28T15:48:26.091Z",
+}
```
Comment (BenRamchandani) [inline tests/test_Integration.py:113]: Is this used?

File: tests/test_view_model.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        Item('43d8', 2, 'Task 2', 'Task 2 not started', datetime.now(), datetime.now(), 'Completed'),
+        Item('43d8', 3, 'Task 3', 'Task 3 in progress', datetime.now(), datetime.now(), 'Completed'),
+        Item('g3d9', 4, 'Task 4', 'Task 4 completed', datetime.now(), datetime.now(), 'Completed'),
+        Item('k3d3', 5, 'Task 5', 'Task 5 not started', datetime.now(), datetime.now(), 'Completed'),
+        Item('53x9', 6, 'Task 6', 'Task 6 completed', datetime.now(), datetime.now(), 'Completed'),
+    ]
+
+    view_model = ViewModel(items)
+
+    assert view_model.should_show_all_completed_items is True
```
Comment (BenRamchandani) [inline tests/test_view_model.py:52]: It would be nice to also test that the inverse is true with six items

**Review by BenRamchandani — APPROVED** [review]
This looks great, I've left a few comments below, but they're all really minor, go ahead and merge when you're ready.

**Pull request merged**